pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -1006,9 +1259,131 @@ func (r *ReconcileClusterPool) createCloudBuilder(pool *hivev1.ClusterPool, logg
 		cloudBuilder.Region = platform.Azure.Region
 		cloudBuilder.CloudName = platform.Azure.CloudName
 		return cloudBuilder, nil
-	// TODO: OpenStack, VMware, and Ovirt.
+	case platform.OpenStack != nil:
+		credsSecret, err := r.getCredentialsSecret(pool, platform.OpenStack.CredentialsSecretRef.Name, logger)
+		if err != nil {
+			return nil, err
+		}
+		cloudBuilder := clusterresource.NewOpenStackCloudBuilderFromSecret(credsSecret)
+		cloudBuilder.Cloud = platform.OpenStack.Cloud
+		return cloudBuilder, nil
+	// TODO: VMware, and Ovirt.
 	default:
 		logger.Info("unsupported platform")
 		return nil, errors.New("unsupported platform")
 	}
 }
+
+// INFO: [Fairness](https://github.com/openshift/hive/blob/master/docs/enhancements/clusterpool-inventory.md#fairness)
+//       The function loops over the list of inventory items and picks the first available customization.
+//		 Failing to apply a customization (in any cluster pool) will cause to change its status to unvailable and a new cluster will be queued.
+func (r *ReconcileClusterPool) getInventoryCustomization(pool *hivev1.ClusterPool, logger log.FieldLogger) (*hivev1.ClusterDeploymentCustomization, error) {
```

Comment:
So yeah, I think we can move the call to this function down a couple lines, add the CD name as a param, and update the CD ref and the Available condition at the same time.